client/src/components/CKANPresentedOptions.js

```diff
@@ -0,0 +1,581 @@
+   *
+   * @return {Row}
+   */
+  renderCheckboxListAndSeparator() {
+    // Don't render the checkbox list unless we've chosen to select from a list of options
+    if (this.getSelectType() !== SELECT_TYPE_ALL) {
+      return null;
+    }
+
+    return (
+      <Row>
+        <Col lg={9} sm={12}>{ this.renderCheckboxList() }</Col>
+        <Col lg={3} sm={12}>{ this.renderSeparator() }</Col>
+      </Row>
+    );
+  }
+
+  /**
+   * Renders a chose for either "select all options" or "manually select options" in a radio
+   * input group
+   *
+   * @returns {FormGroup[]}
+   */
+  renderRadioOptions() {
+    const { data: { selectTypes } } = this.props;
+    const selectedValue = this.getSelectType();
+
+    return selectTypes.map((option) => (
+      <FormGroup key={option.value} className="ckan-presented-options__option-group">
+        <Label for={`option-${option.value}`} check>
+          <Input
+            id={`option-${option.value}`}
+            type="radio"
+            name={this.getFieldName('select-type')}
+            value={option.value}
+            onChange={this.handleSelectTypeChange}
+            checked={selectedValue === String(option.value)}
+          />
+          {option.title}
+        </Label>
+      </FormGroup>
+    ));
+  }
+
+  render() {
+    const { extraClass } = this.props;
+
+    return (
+      <div className={extraClass}>
+        { this.renderRadioOptions() }
+        { this.renderCheckboxListAndSeparator() }
+        { this.renderFreetextInput() }
+        { this.renderHiddenInput() }
+      </div>
+    );
+  }
+}
+
+CKANPresentedOptions.propTypes = {
+  // A list of CKAN fields (aka columns) that this options field will be targeting
+  selectedFields: PropTypes.arrayOf(PropTypes.string),
+  data: PropTypes.shape({
+    // The CKAN endpoint that this field is suggestion options for
+    endpoint: PropTypes.string.isRequired,
+    // The CKAN resource that this field is suggestion options for
+    resource: PropTypes.string.isRequired,
+    selectTypeDefault: PropTypes.string,
+    selectTypes: PropTypes.arrayOf(PropTypes.shape({
+      value: PropTypes.string,
+      title: PropTypes.string,
+    })),
+  }),
+  extraClass: PropTypes.string,
+  name: PropTypes.string,
+  value: PropTypes.object,
+  // TextFieldComponent: PropTypes.oneOfType([
+  //   PropTypes.string,
+  //   PropTypes.func
+  // ]).isRequired,
+  // FormActionComponent: PropTypes.oneOfType([
+  //   PropTypes.string,
+  //   PropTypes.func
+  // ]).isRequired,
+  LoadingComponent: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.func
+  ]).isRequired
+};
+
+CKANPresentedOptions.defaultProps = {
+  data: {},
+  extraClass: '',
+  selectedFields: [],
+  value: {},
+};
+
+export { CKANPresentedOptions as Component };
+
+export default fieldHolder(inject(
+  ['TextField', 'FormAction', 'Loading'],
+  (TextFieldComponent, FormActionComponent, LoadingComponent) => ({
+    TextFieldComponent,
```

Comment:
This and the form action aren't needed any more, only the loading component (since we're using Reactstrap components instead)